docs/ibc/1_IBC_ARCHITECTURE.md

```diff
@@ -0,0 +1,141 @@
+
+### Flow control
+
+IBC does not require specific provision for computation-level flow control since the underlying ledgers will have throughput limitations and flow control mechanisms of their own (such as "gas" markets). Application-level flow control — limiting the rate of particular packets according to their content — may be useful to ensure security properties (limiting the value on a single ledger) and contain damage from Byzantine faults (allowing a challenge period to prove an equivocation, then closing a connection). IBC provides facilities for modules to reject packets and leaves particulars up to the higher-level application protocols.
+
+### Authentication
+
+All datagrams in IBC are authenticated: a block finalized by the consensus algorithm of the sending blockchain must commit to the outgoing datagram via a cryptographic accumulator, and the receiving chain must verify both the consensus transcript and the accumulator proof that the datagram was sent (and associated actions executed) before acting upon it. For correct execution semantics, both chains must implement the correct IBC protocol — of course, since IBC is permissionless, any chain could connect to another chain and provide a correct accumulator proof but incorrectly implement IBC such that its IBC handler did not behave according to protocol, so modules utilizing IBC connections (or end-users using those modules) must inspect the state of the connections and ledgers they plan to use to ensure correct implementation.
+
+### Connections
+
+Reliability, flow control, and authentication as described above require that IBC initializes and maintains certain status information for each datastream. This information is called a connection. Each connection object contains information about the consensus state of the connected ledger, negotiated encoding & multiplexing options, and state & sequence numbers. When two modules wish to communicate, they must locate an existing connection between their two ledgers, or initialize a new connection if none yet exists. Initializing a connection requires a multi-step handshake which, once complete, ensures that only the two intended ledgers are connected and future datagrams relayed will be authenticated, encoded, and sequenced as desired.
+
+### Multiplexing
```

Comment:
Rename "channels"?

Reply:
Channels provide multiplexing.

Reply:
Yes, but I feel it's less explicit. Channels are a core part of the architecture, I think they deserve to appear as section title

Reply:
The section titles are supposed to be about provided properties, I think that makes more sense to a reader since they won't know what channels are because we haven't defined them.

"Connections" also violates this, I changed it to "Statefulness"